openrewrite/src/javascript/parser.ts

```diff
@@ -26,4 +29,1572 @@ export class JavaScriptParser extends Parser {
     sourcePathFromSourceText(prefix: string, sourceCode: string): string {
         return prefix + "/source.js";
     }
+}
+
+export class JavaScriptParserVisitor {
+    visit(sf: ts.SourceFile) {
+        sf.forEachChild((node) => this._visit(node));
+    }
+
+    _visit(node: ts.Node) {
+        var member = this[(`visit${ts.SyntaxKind[node.kind]}` as keyof JavaScriptParserVisitor)];
+        if (typeof member === 'function') {
+            return member(node as any);
+        } else {
+            return this.visitUnknown(node);
+        }
+    }
+
+    visitClassDeclaration(node: ts.ClassDeclaration) {
+        // return new ClassDeclaration(randomId(), node.)
+        return this.visitUnknown(node);
+    }
+
+    visitUnknown(node: ts.Node) {
+        return new J.Unknown(randomId(), Space.EMPTY, Markers.EMPTY, new Source(randomId(), Space.EMPTY, Markers.EMPTY, node.getText()));
+    }
+
+    visitEndOfFileToken(node: ts.Node) {
```

Comment:
Am I correct in assuming that this class is generated? If so, we should also add the generator to the Git repo somewhere, so that we can rerun it when necessary. To that end I propose a few changes:
* Create an interface in a separate file, so that we can actually regenerate it
* Onlt generate a visit method for `Node` subtypes. Specifically, I would probably only generate a visit method for those Node subtypes which have a corresponding `create` method in the `NodeFactory` interface
* Change the parameter type to use the corresponding interface type. E.g. `visitIdentifier(node: ts.Identifier)`
* I think it would probably also be good to make the Visitor interface generic and then have an additional parameter of that generic `T` type, which we can pass around.
* Possibly we should also have a generic type parameter for the return type, which in our case would probably be `J` then. But unsure about that part.

Reply:
it is not. I manually imported all possible methods

Reply:
we can remove method that are not used after we start working on lst. all those methods are created based on all possible kinds of nodes (see ts.SyntaxKind for more info)